source/core.lisp

```diff
@@ -110,8 +102,10 @@ For each instruction it writes the result back to it."
              (destroy ()
                :report "Destroy the existing socket."
                (uiop:delete-file-if-exists (electron-socket-path interface)))))
+         (uiop:chdir (asdf:system-relative-pathname :cl-electron "source"))
          (setf (process interface)
-               (uiop:launch-program `("electron"
+               (uiop:launch-program `("npx"
```

Comment:
Same as above, the way the launch function resolves the path of electron is something that I don't like. Theoretically you could install `cl-electron` in your path and avoid all of this manual path resolution, but that is a different problem. Perhaps that should be added to `npm install`.

Reply:
Said another way, we could just do `npm install -g` in the root of the source directory and then just call `cl-electron`

source: https://docs.npmjs.com/downloading-and-installing-packages-globally

Reply:
It is certainly a bad idea to rely on `npx` since it is not meant to be used for batch processing. On a clean install you'd see:

```log
$ npx electron
Need to install the following packages:
  electron@30.0.1
Ok to proceed? (y) 
```

Reply:
I ran `npm i` from `cl-electron/source` followed by:

```log
~/common-lisp/cl-electron/source$ npx electron
[41165:0426/145333.487764:FATAL:setuid_sandbox_host.cc(158)] The SUID sandbox helper binary was found, but is not configured correctly. Rather than run without sandboxing I'm aborting now. You need to make sure that /home/aadcg/common-lisp/cl-electron/source/node_modules/electron/dist/chrome-sandbox is owned by root and has mode 4755.
/home/aadcg/common-lisp/cl-electron/source/node_modules/electron/dist/electron exited with signal SIGTRAP
```

After doing as told above it worked, but it would be nice to avoid this step otherwise the call to `uiop:launch-program` will silently not start the app.

Reply:
That's interesting. I didn't have to do that. Did you install Node via APT? I used NVM so that I could pick a specific version.

Reply:
This is actually a known issue, see https://github.com/electron/electron/issues/17972.

No need to worry about it.